src/ray/raylet/worker_pool.cc

```diff
@@ -1177,13 +1186,14 @@ void WorkerPool::PopWorker(const TaskSpecification &task_spec,
       } else {
         state.starting_workers_to_tasks[startup_token] = std::move(task_info);
       }
+    } else if (status == PopWorkerStatus::TooManyStartingWorkerProcesses) {
+      DeleteRuntimeEnvIfPossible(task_spec.SerializedRuntimeEnv());
+      state.pending_pop_worker_requests.emplace_back(
+          PopWorkerRequest{task_spec, callback, allocated_instances_serialized_json});
```

Comment:
Do we need to avoid the memory copy of task_spec? Maybe we can optimize it in future and add a TODO here.

Reply:
I think we need to make a copy here. The caller pass this in as a reference, if we don't make a copy, the caller may destruct the object later on.